src/evict/evict_page.c

```diff
@@ -872,9 +878,20 @@ __evict_reconcile(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags)
          * outside world.
          */
         __wt_txn_bump_snapshot(session);
-    else if (use_snapshot_for_app_thread)
+    else if (use_snapshot_for_app_thread) {
+        /*
+         * Application threads entering into eviction saves the existing snapshots and refresh to
+         * acquire a new snapshot to evict the latest data, once the application threads are done
+         * with eviction then the snapshots are switched back to its original snapshots.
```

Comment:
We should add some comments about the WT_EVICT_CALL_REFRESH_SNAPSHOT flag

Reply:
Good call - I've updated this now.